src/Appendices/QTT.md

```diff
@@ -0,0 +1,477 @@
+variables in the context have quantity 0, since all variables are used
+0 times in a constructor.
+
+### Function Application
+
+```repl
+ qf x : T
+------------------------------
+F : (r _ : A) -> B
+
+ qe x : T
+------------------------------
+E : A
+
+ (qf + r*qe) x : T
+------------------------------
+(F E) : B
+```
+
+This is the most complicated of QTT's rules. We have a function `F`
+whose parameter has `r` quantity, and we're applying it to `E`. If our
+variable `x` is used `qf` times in `F` and `qe` times in `E`, then it
+is used `qf + r*qe` times in the full expression.
+
+To better understand this rule, let's look at some simpler cases.
+First, let's assume that `x` is not used in the function `F`, so that
+`qf = 0`. Then, `x`'s full quantity is `r * qe`. For example, let's
+look at these two functions:
+
+```idris
+f x = id x
+
+g x = id 1
+```
+
+Here, `id` has type `a -> a`, where its input is unrestricted (`ω`).
+In the first function, we can see that `x` is used once in the input
+of `id`, so the quantity of `x` in the whole expression is `ω * 1 = ω`.
+In the second function, `x` is used zero times in the input of
+`id`, so its quantity in the whole expression is `ω * 0 = 0`. The
+function `g` will typecheck if you mark its input as erased, but not
+`f`.
+
+As another simplified case, let's assume that `F` is a linear
+function, meaning that `r = 1`. Then `x`'s full quantity is `qf + qe`,
+the simple sum of the quantities of each part. Here's a function that
+demonstrates this:
+
+```idris
+ldup x = (#) x x
+```
+
+The linear pair constructor `(#)` is linear in both arguments, so to
+find the quantity of `x` in the full expression we can just add up the
+quantities in each part. `x` is used zero times in `(#)` and one time
+in `x`, so the total quantity is `0 + 1 + 1 = ω`. If the second `x`
+were replaced by something else, like a literal, the quantity would
+only be `0 + 1 + 0 = 1`. Intuitively, you can think of these as
+"parallel expressions", and the addition operation tells you how
+quantities combine in parallel.
+
+### Subusaging
+
+```repl
+ q x : T
+------------------------------
+E : T'
+
+(q ≤ r)
+
+ r x : T
+------------------------------
+E : T'
+```
+
+This rule is where the order relation on quantities comes in. It
+allows us to convert a quantity in our context to another one, given
+that the new context is greater than or equal to the old one. Type
+theorists call this *subusaging*, as it lets us use variables less
+often than we claim in our types.
+
+Subusaging is why this function definition is allowed:
+
+```idris
+ignore : a -> Int
+ignore x = 42
+```
+
+The input `x` is used zero times, which would normally mean its
+quantity would have to be `0`; however, since `0 ≤ ω`, we can use
+subusaging to increase the quantity to `ω`.
+
+This also explains the mysterious fact we pointed out earlier, that `0
+≰ 1` in our quantity ordering. If it were true that `0 ≤ 1`, then we
```

Comment:
This should go to the previous line: `0 ≰ 1`, otherwise, the backticks are not properly closed.

Reply:
Ah, thanks for catching that. I usually use soft line wrapping when editing text files and then apply an algorithmic line breaking system afterwards, which can sometimes mess up things.